src/tf_service/persistent_service.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2025 Magazino GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import typing
+
+import rospy
+import rospy.core
+from rospy.exceptions import TransportInitError
+from rospy.impl.tcpros_base import TCPROSTransport
+
+PING_TIMEOUT = 0.1
+
+
+class PersistentService(rospy.ServiceProxy):
+    """
+    Persistent service proxy with extensions that are missing from rospy.
+    """
+
+    def __init__(self, name: str, service_class: typing.Any) -> None:
+        # Re-declaration with typing to make mypy happy:
+        self.transport: typing.Optional[TCPROSTransport] = None
+
+        super().__init__(name, service_class, persistent=True)
+        self.init_transport()
+
+    def is_valid(self) -> bool:
+        """
+        rospy doesn't have an equivalent of ros::ServiceClient::isValid().
+        This should do the trick instead...
+        """
+        if self.transport is None:
+            return False
+
+        # With a persistent service, the self.transport is kept after service calls.
+        # Transport loss is only noticed when something is transmitted.
+        # "Ping" the remote address with a header handshake to check if it's alive.
+        dest_addr, dest_port = self.transport.dest_address
+        try:
+            self.transport.connect(
+                dest_addr, dest_port, self.transport.endpoint_id, timeout=PING_TIMEOUT
```

Comment:
probably most of the time we are on a local network and the ping should be quite fast but do you think that `PING_TIMEOUT` should be a parameter of this class?

Reply:
added